### PR TITLE
Add multi-label text classification support to pytorch example

### PR DESCRIPTION
**Pull request by ranchlai**
# What does this PR do?
The transoformer config supports multi-label classification by setting config.problem_type = "multi_label_classification", 
but the run_glue.py does not support it.
This PR add `run_classification.py` to support multi-label classification task.

Main changes compared to `run_glue.py`:
- [x] Add support for multi-label classification task and datasets, e.g., [Reuters-21578](https://huggingface.co/datasets/reuters21578).
- [x] Remove code related to glue tasks
- [x] Update README.md for multi-label classification task.
- Add parameraters and code to support single/multi-label classification and regression task
    - Add `shuffle_train_dataset` option to shuffle train dataset. This is useful to avoid problems caused by ordered labels.
    - Add `metric_name` to specify the metric used to evaluate the model. 
    - Add `remove_splits` to remove some unnsed splits from the dataset, e.g., Reuter dataset has "unused" split, IMDB dataset has "unsupervised" split.
    - Add `remove_columns` to remove some unnsed columns from the dataset
    - Add `text_column_names` to specify the (possibly multiple) columns containing the text.
    - Add `label_column_name` to specify the column containing the labels, e.g.,  `stars" for amazon review dataset
    - Add train/validation/test_split_name to specify the split name for train/validation/test dataset
    - Add do_regression to force treating text-classification task as regression task. This remove the need to change the label dtype of the dataset.

### CODE REVIEW

**Comment by amyeroberts** [comment]
@ranchlai Thanks for opening this PR and contributing to the examples! 

Could you add to the README for this example a snippet for running on a multi-label classification task?

**Comment by ranchlai** [comment]
> @ranchlai Thanks for opening this PR and contributing to the examples!
> 
> Could you add to the README for this example a snippet for running on a multi-label classification task?

Sure^_^.  Working on the  [reuters21578](https://huggingface.co/datasets/reuters21578) dataset  as a minimum example. Will update README accordingly.

**Comment by sgugger** [comment]
@ranchlai Please note that the examples are kept simple to be more readable. This adds a lot of complexity to the original example for something that is not covered by the primary goal of that example (run GLUE benchmark) so I would keep it separate.

**Comment by ranchlai** [comment]
Thanks for commenting @sgugger. I understand and that's why I am trying to make the change minimum.  Multiple label classification is indeed more complicated . Hence, adding a demo in the "text-classification" example could be helpful. Thank you~

**Comment by sgugger** [comment]
Yes, but maybe it could go in a new file focused on text classification only (and not GLUE)?

**Comment by ranchlai** [comment]
That's a good idea. How about run_classification.py in parallel to run_glue.py?  I can try to work it out.

**Comment by sgugger** [comment]
Perfect!

File: examples/pytorch/text-classification/run_classification.py [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+        label_list.sort()
+        num_labels = len(label_list)
+
+    # Load pretrained model and tokenizer
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        finetuning_task=None,
```
Comment (sgugger) [inline examples/pytorch/text-classification/run_classification.py:407]: Here there is a `problem_type` argument you should use to set the right task (regression, single label or multi label). This will make sure the model fine-tuned/trianed works properly in a pipeline after.

File: examples/pytorch/text-classification/run_classification.py [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+    if is_multi_label:
+        config.problem_type = "multi_label_classification"
+        logger.info("setting problem type to multi label classification")
```
Comment (sgugger) [inline examples/pytorch/text-classification/run_classification.py:414]: Should be also set to regression if it's a regression.

**Review by sgugger — COMMENTED** [review]
Thanks a lot for iterating! I just have a couple of comments around the problem type.

File: examples/pytorch/text-classification/run_classification.py (hunk 2 above)
Comment (ranchlai) [reply examples/pytorch/text-classification/run_classification.py]: Fixed. Thanks for reviewing

File: examples/pytorch/text-classification/run_classification.py (hunk 1 above)
Comment (ranchlai) [reply examples/pytorch/text-classification/run_classification.py:471]: I have set "text-regression", "text-classification" and "multi-label-text-classification" for the three finetuning tasks. Is it ok? please would you kindly review~

**Comment by ranchlai** [comment]
@sgugger please would you leave more comments, although I am still running more tests

File: examples/pytorch/text-classification/run_classification.py (hunk 1 above)
Comment (ranchlai) [reply examples/pytorch/text-classification/run_classification.py:471]: Note that I use "-" instead of "_" to follow the code in other examples such as [https://github.com/huggingface/transformers/blob/129cb6d5239374869beea0e1317641426e8df572/examples/pytorch/image-classification/run_image_classification_no_trainer.py#L300](image-classification)

Comment (sgugger) [reply examples/pytorch/text-classification/run_classification.py:468]: I think for the fine-tuning task those are not the labels expected by the Hub. It should be `text-classification` in all cases.

Comment (ranchlai) [reply examples/pytorch/text-classification/run_classification.py]: yes,  I have fixed it.

**Review by sgugger — APPROVED** [review]
Thanks again for adding this! Are you all good with your tests, can we merge this?

**Comment by ranchlai** [comment]
> 

I think I have finished my tests. Scripts [here](https://github.com/ranchlai/transformers/tree/add_test_scripts/examples/pytorch/text-classification/test]). Please merge if looks good.

**Comment by sgugger** [comment]
Thanks again for your contribution!

**Comment by happypigzy** [comment]
I think the added content in README should be placed in the bottom.
